backend/src/API/teamEventsAPI.ts

```diff
@@ -8,53 +8,71 @@ const teamEventAttendanceDao = new TeamEventAttendanceDao();
   const canEditTeamEvents = await PermissionsManager.canEditTeamEvent(user);
   if (!canEditTeamEvents) throw new PermissionError('does not have permissions');
-  return TeamEventsDao.getAllTeamEvents();
+  const teamEvents = await TeamEventsDao.getAllTeamEvents();
+  return Promise.all(
+    teamEvents.map(async (event) => ({
+      ...event,
+      attendees: (await teamEventAttendanceDao.getTeamEventAttendanceByEventId(event.uuid)).filter(
+        (attendance) => attendance.pending === false
+      ),
+      requests: (await teamEventAttendanceDao.getTeamEventAttendanceByEventId(event.uuid)).filter(
+        (attendance) => attendance.pending === true
+      )
+    }))
+  );
 };
 
 export const getAllTeamEventInfo = async (): Promise<TeamEventInfo[]> =>
   TeamEventsDao.getAllTeamEventInfo();
 
 export const createTeamEvent = async (
-  teamEvent: TeamEvent,
+  teamEventInfo: TeamEventInfo,
   user: IdolMember
-): Promise<TeamEvent> => {
+): Promise<TeamEventInfo> => {
   const canCreateTeamEvent = await PermissionsManager.canEditTeamEvent(user);
   if (!canCreateTeamEvent)
     throw new PermissionError('does not have permissions to create team event');
-  await TeamEventsDao.createTeamEvent(teamEvent);
-  return teamEvent;
+  await TeamEventsDao.createTeamEvent(teamEventInfo);
+  return teamEventInfo;
 };
 
 export const deleteTeamEvent = async (teamEvent: TeamEvent, user: IdolMember): Promise<void> => {
   if (!PermissionsManager.canEditTeamEvent(user)) {
     throw new PermissionError("You don't have permission to delete a team event!");
   }
+  const attendances = teamEvent.attendees;
+  await Promise.all(
+    attendances.map((attendance) =>
+      teamEventAttendanceDao.deleteTeamEventAttendance(attendance.uuid)
+    )
+  );
   await TeamEventsDao.deleteTeamEvent(teamEvent);
 };
 
 export const updateTeamEvent = async (
-  teamEvent: TeamEvent,
+  teamEventInfo: TeamEventInfo,
   user: IdolMember
-): Promise<TeamEvent> => {
+): Promise<TeamEventInfo> => {
   if (!PermissionsManager.canEditTeamEvent(user)) {
     throw new PermissionError(
       `User with email ${user.email} does not have permissions to update team events`
     );
   }
-  await TeamEventsDao.updateTeamEvent(teamEvent);
-  return teamEvent;
+  const updatedTeamEvent = await TeamEventsDao.updateTeamEvent(teamEventInfo);
+  return updatedTeamEvent;
 };
 
 export const requestTeamEventCredit = async (
-  uuid: string,
-  request: TeamEventAttendance
+  request: TeamEventAttendance,
+  user: IdolMember
 ): Promise<void> => {
-  const teamEvent = await TeamEventsDao.getTeamEvent(uuid);
-  const updatedTeamEvent = {
-    ...teamEvent,
-    requests: [...teamEvent.requests, request]
-  };
-  await TeamEventsDao.updateTeamEvent(updatedTeamEvent);
+  if (user.email !== request.member.email) {
+    throw new PermissionError(
+      `User with email ${user.email} does not have permissions to request this team event credit`
```

Comment:
I would change this to "User with email ${user.email} cannot request team event credit for another member, ${request.member.email}." In the event that something wonky happens, this is a more clear error message.